lib/worker.js

```diff
@@ -150,6 +155,23 @@ async function updateDependency(upgrade) {
       newVersion);
   }
 
+  function getYarnLockFileName(packageFileName) {
+    const packagePath = packageFileName.split('/');
+    // Remove 'package.json'
+    const basePath = packagePath.slice(0, -1);
+    basePath.push('yarn.lock');
+    return basePath.join('/');
+  }
+
+  function generateLockFile(newPackageJson) {
+    logger.debug('Generating new yarn.lock file');
+    rimraf.sync('/tmp/renovate-yarn');
+    mkdirp.sync('/tmp/renovate-yarn');
+    fs.writeFileSync('/tmp/renovate-yarn/package.json', newPackageJson);
+    cp.spawnSync('yarn', ['install'], { cwd: '/tmp/renovate-yarn' });
```

Comment:
For this to work we'd also need the `.npmrc` file from the repo if it exists so that auth tokens or custom registries work.